pkg/plugins/policies/meshtls/plugin/v1alpha1/plugin.go

```diff
@@ -186,19 +182,19 @@ func applyToRealResources(
 ) error {
 	for _, resType := range rs.IndexByOrigin(core_xds.NonMeshExternalService) {
 		// there is only one rule always because we're in `Mesh/Mesh`
-		var conf *api.Conf
-		for _, r := range fromRules.Rules {
-			conf = core_rules.ComputeConf[api.Conf](r, subsetutils.MeshElement())
+		var conf api.Conf
+		for _, r := range fromRules.InboundRules {
+			conf = rules_inbound.MatchesAllIncomingTraffic[api.Conf](r)
 			break
 		}
-		if conf == nil {
-			continue
-		}
 
 		for typ, resources := range resType {
 			switch typ {
 			case envoy_resource.ClusterType:
 				for _, cluster := range resources {
+					if cluster.Origin == metadata.OriginGateway {
```

Comment:
Is this change caused by `spec.rules` support or by something else?

Reply:
yes, this is caused by `spec.rules` previously we were skipping this on nil conf, now conf is not nill, so we should not configure gateway resources here

Reply:
can you filter out gateway resources above?
```go
for _, resType := range rs.IndexByOrigin(core_xds.NonMeshExternalService) {
```

Reply:
changed